server/segment_rpc.go

```diff
@@ -328,6 +330,18 @@ func SubmitSegment(sess *BroadcastSession, seg *stream.HLSSegment, nonce uint64)
 		return nil, err
 	}
 
+	ratPrice, _ := ratPriceInfo(sess.OrchestratorInfo.PriceInfo)
+	faceVal := new(big.Int).SetBytes(sess.OrchestratorInfo.GetTicketParams().GetFaceValue())
+	winProb := new(big.Int).SetBytes(sess.OrchestratorInfo.GetTicketParams().GetWinProb())
+	winProbRat := new(big.Rat).Mul(new(big.Rat).SetFrac(winProb, pm.MaxWinProb), big.NewRat(100, 1)).FloatString(10) + " %"
+	glog.V(common.VERBOSE).Infof("Created new payment - recipient=%v  faceValue=%v  winProb=%v  price=%v  numTickets=%v",
```

Comment:
What about placing this log statement in `genPayment()` within the code block that executes if `numTickets > 0`? This would also resolve the current CI failure related to the lack of an on-chain mode check here since `genPayment()` already contains an on-chain mode check.

Reply:
Yup that's true, I guess I didn't like that we would log before assessing of marshalling the protobuf should yield an error but in hindsight this should never really error.

Reply:
fixed in 5a6081e